node/hyperbahn/service_proxy.js

```diff
@@ -548,6 +539,25 @@ function destroy() {
     self.rateLimiter.destroy();
 };
 
+ServiceDispatchHandler.prototype.initCircuits =
+function initCircuits() {
+    var self = this;
+
+    self.circuits = new Circuits({
+        timeHeap: self.channel.timeHeap,
+        timers: self.channel.timers,
+        random: self.random,
+        egressNodes: self.egressNodes,
+        config: self.circuitsConfig
+    });
+
+    self.circuits.circuitStateChangeEvent.on(onCircuitStateChange);
+
+    function onCircuitStateChange(stateChange) {
```

Comment:
why bother to have a closure? can we do self.circuits.circuitStateChangeEvent.on(self.onCircuitStateChange); ?